zipline/finance/performance/tracker.py

```diff
@@ -189,7 +199,9 @@ def get_portfolio(self, performance_needs_update):
         if performance_needs_update:
             self.update_performance()
             self.account_needs_update = True
-        return self.cumulative_performance.as_portfolio()
+        portfolio = self.cumulative_performance.as_portfolio()
+        portfolio.current_date = self._current_session
```

Comment:
I don't think it's guaranteed that `get_portfolio` will called every day.  Even if that's true now, I wouldn't necessarily expect that to be the case in the future.  If the portfolio needs to know the current date, then I think we want an explicit documented API that includes a contract for when the date should be updated and whose responsible for performing that update.